frontend/app/controllers/spree/coupon_codes_controller.rb

```diff
@@ -18,7 +20,7 @@ def create
               flash[:success] = handler.success
               redirect_to cart_path
             else
-              flash.now[:error] = handler.error
+              flash.now[:error] = safe_join(handler.errors.full_messages, " <br /> ".html_safe)
```

Comment:
What about providing a generic coupon failure message and also set a `@coupon_handler` instance variable that can be used in the view layer to retrieve the errors? I'm just concerned that some store flash implementation is not ready to have multi-lines messages.

Reply:
Great idea! I'll try to do it a bit later.